### ISSUE TITLE
`sprintf "%b" bool` causes AOT failure

### ISSUE THREAD

**Issue by Smaug123** [dotnet/fsharp#17691]
`sprintf "%b" true` does not work in AOT context; nor does `sprintf "%O" true`. This affects linux-x64 8.0.401 and v9 preview 1, but not osx-arm64 8.0.300, in my testing.

**Repro steps**

In an executable with `<PublishAot>true</PublishAot>`:

```fsharp
let s = sprintf "%O" true // or "%b"
System.Console.WriteLine s
```

**Expected behavior**

Program prints "true".

**Actual behavior**

```
System.NotSupportedException: 'Microsoft.FSharp.Core.PrintfImpl+Specializations`3[Microsoft.FSharp.Core.Unit,System.String,System.String].CaptureFinal1[System.Boolean](Microsoft.FSharp.Core.PrintfImpl+Step[])' is missing native code.
```
```

**Known workarounds**

`(fun s -> $"%O{s}") true` works.

**Related information**

Reproduced on linux-x64 with net8.0.401 and net9 preview 7, but *not* osx-arm64 8.0.300.

**Comment by charlesroddie** [comment]
Do you have need for sprintf in preference to interpolated strings? It would be better to focus effort on improving interpolated strings (by making them work via a string collector) than working on sprintf which is generally unsalvageable.

**Comment by Smaug123** [comment]
Yeah, I write source generators and am not really willing to break compatibility. I could of course parameterise the generators or do some work to make available to them the current version of F#, or more practically I could just accept that there are bugs and manually call ToString on everything; that's certainly not the end of the world.

**Comment by charlesroddie** [comment]
F# 5.0 with string interpolation was released 2020. Can't imagine why anyone would be using previous versions at this point.

Anyway the options are:
- sprintf: bad performance, AOT-incompatible, medium readability.
- interpolated strings: bad performance, AOT-compatible, high readability. Planned to be made good performance in future with string collectors (and likely less bloat from the current implementation which involves v complex intermediate types).
- manually calling ToString and concatenating the results: good performance, AOT compatible, low readability.
